### ISSUE TITLE
tty related problem after upgrading from 17.03.2 when using gnu-screen

### ISSUE THREAD

**Issue by bugdanov** [docker/for-linux#111]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior

stdio and signals should be coming from and going to the container, when starting a container with ```screen -dm docker run -it ...```

### Actual behavior
I do not see stdout, and ctrl-c does not stop the container

### Steps to reproduce the behavior

Everything below was working fine with docker-ce 17.03.2 build f5ec1e2 on debian stretch, and broke after upgrading to 17.06.2-ce, build cec0b72

Using Screen version 4.05.00 (GNU) 10-Dec-16

```
screen -S test -dm docker run --rm=true --name test -it alpine sh
screen -r test
# Nothing is displayed on stdout
# Now after pressing <ctrl-c>,  the container should be stopped
# But it is not:
docker rm -f test
```
**Output of `docker version`:**

```
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 20:00:06 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:58:57 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 51
Server Version: 17.06.2-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 19
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-3-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 47.15GiB
Name: z600
ID: ZXBI:VXZQ:2QNO:IHWF:ZLYX:DRDZ:M4EO:5HNS:YN45:ILRI:QDRY:XEE6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

**Comment by seemethere** [comment]
Could not recreate with `17.09.0-ce` on Debian Stretch, would it be possible for you to upgrade?

[![asciicast](https://asciinema.org/a/1BJXEk2bbEIlMRgdggqUndato.png)](https://asciinema.org/a/1BJXEk2bbEIlMRgdggqUndato)

**Comment by bugdanov** [comment]
Problem solved with 17.09.0~ce-0~debian
